src/Nerdbank.MessagePack/ByValueEqualityComparer`2.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Andrew Arnott. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Nerdbank.MessagePack.SecureHash;
+
+namespace Nerdbank.MessagePack;
+
+/// <summary>
+/// Provides deep by-value implementations of <see cref="IEqualityComparer{T}"/> for arbitrary data types.
+/// </summary>
+/// <typeparam name="T">The data type to be hashed or compared by value.</typeparam>
+/// <typeparam name="TProvider">The provider of the type shape.</typeparam>
+/// <remarks>
```

Comment:
@eiriktsarpalis If you study nothing else, I'd appreciate you studying the policies as documented in this remarks section and sending me your feedback, please.